src/services/storage/handlers/offer.ts

```diff
@@ -46,26 +47,28 @@ const handler: Handler<StorageServices> = {
           break
         }
 
-        const [firstMsg, ...restMsg] = msg[0].replace('0x', '')
-        const flag = firstMsg.substring(0, 2)
+        const [firstMsg, ...restMsg] = msg
+        const flag = firstMsg.substring(2, 4)
 
         if (flag === '01') { // PeerId definition
-          offer.peerId = decodeByteArray([firstMsg.substring(2), ...restMsg])
+          offer.peerId = decodeByteArray([`0x${firstMsg.substring(4)}`, ...restMsg])
           logger.info(`PeerId ${offer.peerId} defined (ID: ${offer.address})`)
         } else {
-          logger.error(`Unknown message flag ${flag}!`)
+          throw new EventError(`Unknown message flag ${flag}!`, event.event)
         }
         break
       }
       case 'BillingPlanSet':
         await updatePrices(offer, event.returnValues.period, event.returnValues.price)
         break
       default:
-        logger.error(`Unknown event ${event.event}`)
```

Comment:
So basically this is a dead code as we have an event filtering [here](https://github.com/rsksmart/rif-marketplace-cache/blob/master/src/services/storage/storage.processor.ts#L14) and all unknown events already ignored.
Should I add some handler for unknown events?

Reply:
Hmmm I see, you are right. Yet even though in runtime this might not happen then based on input data from the blockchain, for example during development there might be some inconsistency in configuration and lead to this? I admit that it is not so strong case, but even that I would still keep it there 😅

Reply:
Yes, don't think about configuration. It's reverted 👍